readthedocs/proxito/views/serve.py

```diff
@@ -265,6 +265,14 @@ def _get_canonical_redirect_type(self, request):
                 log.debug("Proxito CNAME HTTPS Redirect.", domain=domain.domain)
                 return RedirectType.http_to_https
 
+        if (
+            unresolved_domain.is_from_public_domain
+            or unresolved_domain.is_from_external_domain
+        ):
+            if settings.PUBLIC_DOMAIN_USES_HTTPS and not request.is_secure():
+                # Redirect HTTP -> HTTPS (302) for public domains.
+                return RedirectType.http_to_https
+
```

Comment:
I think we should probably move the https redirects to the middleware, since we want all traffic to redirect to https, not just doc pages.

Reply:
Doesn't Django ship with middleware to do this? 

https://docs.djangoproject.com/en/4.1/ref/settings/#std-setting-SECURE_SSL_REDIRECT

Reply:
But we need to take custom domains into consideration, we can't redirect everything

Reply:
or at leas in .org, in .com we were redirecting all custom domains.

Reply:
I think now that we have automatic SSL configuration, we should be able to auto-redirect to SSL everywhere? I think we probably want to migrate to only serving on SSL, so this seems like a reasonable step to take? But I agree, we want to make sure we're taking into account the `domain.ssl` variable so we don't break things in the transition.

Reply:
We can probably inherit from the Django one and modify it a little to consider `domain.ssl`. I like re-using starndard Django stuffs :)